### PR TITLE
[macos] use unxip if available for XCode unpacking

### PR DESCRIPTION
**Pull request by ilia-shipitsin**
# Description

small improvement, use "unxip" utility from Homebrew if available

#### Related issue:

https://github.com/actions/runner-images-internal/issues/5272

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

**Review by erik-bershel — APPROVED** [review]
LGTM

File: images/macos/helpers/Common.Helpers.psm1 [hunk 1]
```diff
@@ -156,3 +156,21 @@ function Add-EnvironmentVariable {
...
+ param
+ (
+     $command
+ )
+
+ $oldPreference = $ErrorActionPreference
+ $ErrorActionPreference = 'Stop'
+
+ try {
+     if(Get-Command $command){ $true }
```
Comment (vpolikarpov-akvelon) [inline images/macos/helpers/Common.Helpers.psm1:170]: Can't we use `-ErrorAction SilentlyContinue` and leave `$ErrorActionPreference` variable alone?

Comment (ilia-shipitsin) [reply images/macos/helpers/Common.Helpers.psm1:170]: well, I have no particular opinion on that, I'll use your suggestion after rebase

File: images/macos/helpers/Common.Helpers.psm1 [hunk 2]
```diff
@@ -166,3 +166,12 @@ function Get-Architecture {
 
     return $arch
 }
+
+Function Test-CommandExists {
```
Comment (maxim-lobanov) [inline images/macos/helpers/Common.Helpers.psm1:170]: nit: `function` (lower cases) to be consistent with other functions in this helper

File: images/macos/helpers/Common.Helpers.psm1 [hunk 3]
```diff
@@ -166,3 +166,12 @@ function Get-Architecture {
...
+    param
+    (
+        $command
```
Comment (maxim-lobanov) [inline images/macos/helpers/Common.Helpers.psm1:173]: nit: `[Parameter(Mandatory)] [string] $Command,`
(mandatory attribute, var type and capitalize variable name to be consistent with other functions

**Review by maxim-lobanov — DISMISSED** [review]

File: images/macos/helpers/Common.Helpers.psm1 (hunk 2 above)
Comment (ilia-shipitsin) [reply images/macos/helpers/Common.Helpers.psm1]: fixed

File: images/macos/helpers/Common.Helpers.psm1 (hunk 3 above)
Comment (ilia-shipitsin) [reply images/macos/helpers/Common.Helpers.psm1]: fixed

**Review by maxim-lobanov — APPROVED** [review]

**Review by mikhailkoliada — APPROVED** [review]

**Pull request merged**